plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -416,6 +418,39 @@ public void checkAndSetLatestUpdateCenter() {
             }
         }
     }
+    /**
+     * Outputs a list of plugins which have failed to download
+     */
+    public void outputFailedPlugins() {
+        if (failedPlugins.size() > 0) {
+            System.out.println("Some plugins failed to download: ");
+            //Create a new Failed Plugins text file
+            File failedPluginsFile = new File(refDir.getAbsolutePath() + File.separator + "failed-plugins.txt");
+
+            if (!failedPluginsFile.exists()) {
+                try {
+                    failedPluginsFile.createNewFile();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                    System.out.println("File cannot be created at the path: " + failedPluginsFile
+                        .getAbsolutePath());
+                }
+            }
+
+            try {
+                String newLine = System.getProperty("line.separator");
+                FileWriter fileWriter = new FileWriter(failedPluginsFile.getPath());
+                for (Plugin plugin : failedPlugins) {
+                    String failedPluginName = plugin.getName();
+                    fileWriter.write(failedPluginName + newLine);
+                    System.out.println(failedPluginName);
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+        System.exit(1);
```

Comment:
Why? this is in the library code, we shouldn't be calling system.exit here

Reply:
https://github.com/stopalopa/plugin-installation-manager-tool/blob/e31abdc5ce80e8ca95237c2c5b21fea8100a31cb/plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java#L168
I picked it up from here, I thought it was essential to exiting.

Reply:
hmm k, we shouldn't be doing that, but the library needs a good amount of work to be actually usable as a library

Reply:
okay :+1: